In [3]:
import pandas as pd
path_folder = "/content/drive/MyDrive/Colab Notebooks/2° Cuatri Python/Archivos varios"

## Ejercicio 1

In [4]:
alumnos = {
    "Nombre" : ["María", "Juan", "Pablo", "Luz", "Manuel", "Lara", "Rafael", "Malena"],
    "Legajo" : [1001, 1002, 1003, 1004, 1005, 1006, 1007, 1008],
    "Nota 1" : [8, 7, 9, 6, 10, 9, 8, 9],
    "Nota 2" : [6, 9, 8, 7, 9, 10, 3, 7]
}
df_alumnos = pd.DataFrame(alumnos)
df_alumnos

Nombre  Legajo  Nota 1  Nota 2
0   María    1001       8       6
1    Juan    1002       7       9
2   Pablo    1003       9       8
3     Luz    1004       6       7
4  Manuel    1005      10       9
5    Lara    1006       9      10
6  Rafael    1007       8       3
7  Malena    1008       9       7

## Ejercicio 2

In [5]:
df_alumnos_promedio = df_alumnos.copy()

df_alumnos_promedio["Promedio"] = (df_alumnos["Nota 1"] + df_alumnos["Nota 2"]) / 2
df_alumnos

Nombre  Legajo  Nota 1  Nota 2
0   María    1001       8       6
1    Juan    1002       7       9
2   Pablo    1003       9       8
3     Luz    1004       6       7
4  Manuel    1005      10       9
5    Lara    1006       9      10
6  Rafael    1007       8       3
7  Malena    1008       9       7

In [6]:
df_alumnos_promedio = df_alumnos_promedio.sort_values("Promedio", ascending=False)
df_alumnos_promedio

Nombre  Legajo  Nota 1  Nota 2  Promedio
4  Manuel    1005      10       9       9.5
5    Lara    1006       9      10       9.5
2   Pablo    1003       9       8       8.5
1    Juan    1002       7       9       8.0
7  Malena    1008       9       7       8.0
0   María    1001       8       6       7.0
3     Luz    1004       6       7       6.5
6  Rafael    1007       8       3       5.5

In [7]:
df_mayor_60 = df_alumnos_promedio[df_alumnos_promedio["Promedio"] > 6]
df_mayor_60

Nombre  Legajo  Nota 1  Nota 2  Promedio
4  Manuel    1005      10       9       9.5
5    Lara    1006       9      10       9.5
2   Pablo    1003       9       8       8.5
1    Juan    1002       7       9       8.0
7  Malena    1008       9       7       8.0
0   María    1001       8       6       7.0
3     Luz    1004       6       7       6.5

In [8]:
df_menor_60 = df_alumnos_promedio[df_alumnos_promedio["Promedio"] <= 6]
df_menor_60

Nombre  Legajo  Nota 1  Nota 2  Promedio
6  Rafael    1007       8       3       5.5

## Ejercicio 3

In [9]:
alumnos_agregados = {
    "Nombre" : ["Carlos", "Laura"],
    "Legajo" : [1009, 1010],
    "Nota 1" : [9, 8],
    "Nota 2" : [7, 9]
}

df_alumnos_agregados = pd.DataFrame(alumnos_agregados)
df_alumnos_agregados

Nombre  Legajo  Nota 1  Nota 2
0  Carlos    1009       9       7
1   Laura    1010       8       9

In [10]:
df_alumnos = pd.concat([df_alumnos, df_alumnos_agregados], ignore_index = True)
df_alumnos

Nombre  Legajo  Nota 1  Nota 2
0   María    1001       8       6
1    Juan    1002       7       9
2   Pablo    1003       9       8
3     Luz    1004       6       7
4  Manuel    1005      10       9
5    Lara    1006       9      10
6  Rafael    1007       8       3
7  Malena    1008       9       7
8  Carlos    1009       9       7
9   Laura    1010       8       9

In [11]:
df_alumnos.drop([1,6,7] , inplace = True)
df_alumnos

Nombre  Legajo  Nota 1  Nota 2
0   María    1001       8       6
2   Pablo    1003       9       8
3     Luz    1004       6       7
4  Manuel    1005      10       9
5    Lara    1006       9      10
8  Carlos    1009       9       7
9   Laura    1010       8       9

In [12]:
df_alumnos.reset_index(drop=True, inplace=True)
df_alumnos

Nombre  Legajo  Nota 1  Nota 2
0   María    1001       8       6
1   Pablo    1003       9       8
2     Luz    1004       6       7
3  Manuel    1005      10       9
4    Lara    1006       9      10
5  Carlos    1009       9       7
6   Laura    1010       8       9

## Ejercicio 4

In [13]:
df_ejerc_4= pd.read_csv(path_folder + "/nacimientos_muertes_incrementos.csv")
df_nac_muertes_increm = df_ejerc_4.copy()
df_nac_muertes_increm

December year  Live births  Deaths  Natural increase
0            1952        51846   18897             32949
1            1953        51888   18354             33534
2            1954        54057   18876             35178
3            1955        55596   19224             36372
4            1956        56532   19698             36834
..            ...          ...     ...               ...
65           2017        59610   33339             26268
66           2018        58020   33225             24795
67           2019        59637   34260             25377
68           2020        57573   32613             24960
69           2021        58659   34932             23727

[70 rows x 4 columns]

In [14]:
df_nac_muertes_increm.drop(["Natural increase"], axis=1, inplace=True)
df_nac_muertes_increm

December year  Live births  Deaths
0            1952        51846   18897
1            1953        51888   18354
2            1954        54057   18876
3            1955        55596   19224
4            1956        56532   19698
..            ...          ...     ...
65           2017        59610   33339
66           2018        58020   33225
67           2019        59637   34260
68           2020        57573   32613
69           2021        58659   34932

[70 rows x 3 columns]

In [15]:
df_mas_nacimientos = df_nac_muertes_increm.sort_values("Live births", ascending=False).head(10)
df_mas_nacimientos.reset_index(drop=True, inplace=True)
df_mas_nacimientos

December year  Live births  Deaths
0           1961        65391   21780
1           1962        65013   22083
2           1963        64527   22416
3           1971        64458   24309
4           2008        64341   29187
5           2007        64044   28521
6           2010        63897   28437
7           1972        63216   24801
8           1960        62778   20892
9           2009        62541   28965

In [16]:
df_menos_muertes = df_nac_muertes_increm.sort_values("Deaths",).head(10)
df_menos_muertes.reset_index(drop=True, inplace=True)
df_menos_muertes

December year  Live births  Deaths
0           1953        51888   18354
1           1954        54057   18876
2           1952        51846   18897
3           1955        55596   19224
4           1956        56532   19698
5           1958        60558   20301
6           1957        58425   20862
7           1960        62778   20892
8           1959        61800   21126
9           1961        65391   21780

In [17]:
df_menos_50k_nacimientos = df_nac_muertes_increm[df_nac_muertes_increm["Live births"] < 50000]
df_menos_50k_nacimientos.reset_index(drop=True, inplace=True)
df_menos_50k_nacimientos

December year  Live births  Deaths
0           1982        49938   25533

In [18]:
promedio_nacimientos= df_nac_muertes_increm["Live births"].mean()
promedio_muertes = df_nac_muertes_increm["Deaths"].mean()
print(f"El promedio de nacimientos es: {round(promedio_nacimientos)}")
print(f"El promedio de muertes es: {round(promedio_muertes)}")

El promedio de nacimientos es: 58134
El promedio de muertes es: 26218


## Ejercicio 5

In [19]:
df_ejerc_5 = pd.read_csv(path_folder + "/precios-en-surtidor-resolucin-3142016.csv")
df_precios_combustible = df_ejerc_5.copy()
df_precios_combustible

indice_tiempo  idempresa           cuit               empresa  \
0           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
1           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
2           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
3           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
4           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
...             ...        ...            ...                   ...   
36717       2024-08       4097  30-70795820-7           ZUSICO S.A.   
36718       2024-09       4097  30-70795820-7           ZUSICO S.A.   
36719       2024-09       4097  30-70795820-7           ZUSICO S.A.   
36720       2024-09       4097  30-70795820-7           ZUSICO S.A.   
36721       2024-09       4097  30-70795820-7           ZUSICO S.A.   

             direccion          localidad     provincia    region  idproducto  \
0      Av. Mosconi 299  LOMAS DEL MIRADOR  BUENOS AIRES  PAMPEANA          19   
1      Av. Mosconi 299  LOMAS DEL MIRADOR  BUENOS AIRES  PAMPEANA          19   
2      Av. Mosconi 299  LOMAS DEL MIRADOR  BUENOS AIRES  PAMPEANA          21   
3      Av. Mosconi 299  LOMAS DEL MIRADOR  BUENOS AIRES  PAMPEANA          21   
4      Av. Mosconi 299  LOMAS DEL MIRADOR  BUENOS AIRES  PAMPEANA           6   
...                ...                ...           ...       ...         ...   
36717  TALCAHUANO 1198      MAR DEL PLATA  BUENOS AIRES  PAMPEANA           6   
36718  TALCAHUANO 1198      MAR DEL PLATA  BUENOS AIRES  PAMPEANA           3   
36719  TALCAHUANO 1198      MAR DEL PLATA  BUENOS AIRES  PAMPEANA           3   
36720  TALCAHUANO 1198      MAR DEL PLATA  BUENOS AIRES  PAMPEANA           2   
36721  TALCAHUANO 1198      MAR DEL PLATA  BUENOS AIRES  PAMPEANA           2   

                               producto  idtipohorario tipohorario  precio  \
0                       Gas Oil Grado 2              2      Diurno  1130.0   
1                       Gas Oil Grado 2              3    Nocturno  1130.0   
2                       Gas Oil Grado 3              2      Diurno  1371.0   
3                       Gas Oil Grado 3              3    Nocturno  1371.0   
4                                   GNC              2      Diurno   429.9   
...                                 ...            ...         ...     ...   
36717                               GNC              3    Nocturno   699.0   
36718  Nafta (premium) de más de 95 Ron              2      Diurno  1416.0   
36719  Nafta (premium) de más de 95 Ron              3    Nocturno  1416.0   
36720   Nafta (súper) entre 92 y 95 Ron              2      Diurno  1178.0   
36721   Nafta (súper) entre 92 y 95 Ron              3    Nocturno  1178.0   

            fecha_vigencia  idempresabandera empresabandera    latitud  \
0      2024-09-07 10:51:00                28           PUMA -34.658476   
1      2024-09-07 10:51:00                28           PUMA -34.658476   
2      2024-09-07 10:52:00                28           PUMA -34.658476   
3      2024-09-07 10:52:00                28           PUMA -34.658476   
4      2024-09-07 10:53:00                28           PUMA -34.658476   
...                    ...               ...            ...        ...   
36717  2024-08-12 14:00:00                28           PUMA -38.031169   
36718  2024-09-01 09:00:00                28           PUMA -38.031169   
36719  2024-09-01 09:00:00                28           PUMA -38.031169   
36720  2024-09-01 09:00:00                28           PUMA -38.031169   
36721  2024-09-01 09:00:00                28           PUMA -38.031169   

        longitud                                            geojson  
0     -58.529443  {"type":"Point","coordinates":[-58.529443,-34....  
1     -58.529443  {"type":"Point","coordinates":[-58.529443,-34....  
2     -58.529443  {"type":"Point","coordinates":[-58.529443,-34....  
3     -58.529443  {"type":"Point","coordinates":[-58.529443,-34....  
4   

In [20]:
df_precios_combustible.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36722 entries, 0 to 36721
Data columns (total 19 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   indice_tiempo     36722 non-null  object 
 1   idempresa         36722 non-null  int64  
 2   cuit              36722 non-null  object 
 3   empresa           36722 non-null  object 
 4   direccion         36722 non-null  object 
 5   localidad         36722 non-null  object 
 6   provincia         36722 non-null  object 
 7   region            29385 non-null  object 
 8   idproducto        36722 non-null  int64  
 9   producto          36722 non-null  object 
 10  idtipohorario     36722 non-null  int64  
 11  tipohorario       36722 non-null  object 
 12  precio            36722 non-null  float64
 13  fecha_vigencia    36722 non-null  object 
 14  idempresabandera  36722 non-null  int64  
 15  empresabandera    36722 non-null  object 
 16  latitud           36622 non-null  float6

In [21]:
#df_resultado = df_precios_combustible[df_precios_combustible['localidad'].str.contains('BAHIA')]
#df_resultado

### Inciso a

Según promedio de precios en provincias

In [52]:
round(df_precios_combustible.groupby("provincia")["precio"].mean(), 2).sort_values(ascending = False)

provincia
CORRIENTES             1270.21
MISIONES               1268.16
CHACO                  1223.27
ENTRE RIOS             1177.91
CATAMARCA              1173.89
SAN LUIS               1138.65
BUENOS AIRES           1114.45
SAN JUAN               1111.53
CORDOBA                1104.60
CAPITAL FEDERAL        1097.68
NEUQUEN                1088.43
FORMOSA                1081.64
SANTIAGO DEL ESTERO    1077.18
SANTA FE               1073.81
RIO NEGRO              1067.16
LA PAMPA               1064.08
SANTA CRUZ             1058.60
SALTA                  1056.49
CHUBUT                 1052.83
TUCUMAN                1041.41
JUJUY                  1033.03
MENDOZA                1000.30
TIERRA DEL FUEGO        983.12
LA RIOJA                910.34
Name: precio, dtype: float64

En promedio, los precios más altos se encuentran las provincias de **Corrientes, Misiones y Chaco**.

Según registros específicos (menos preciso)

In [23]:
df_precios_combustible.sort_values("precio", ascending=False).head(10)

indice_tiempo  idempresa           cuit                  empresa  \
33945       2024-08       8182  30-70990544-5             S.J.SPORT SA   
33944       2024-08       8182  30-70990544-5             S.J.SPORT SA   
33942       2024-07       8182  30-70990544-5             S.J.SPORT SA   
33943       2024-07       8182  30-70990544-5             S.J.SPORT SA   
11254       2024-09      10567  30-71735577-2  EL EMPALME EL PINGO SRL   
27485       2024-09       3399  33-69534030-9    PEGORIN Y CÍA. S.R.L.   
27486       2024-09       3399  33-69534030-9    PEGORIN Y CÍA. S.R.L.   
27488       2024-09       3399  33-69534030-9    PEGORIN Y CÍA. S.R.L.   
27487       2024-09       3399  33-69534030-9    PEGORIN Y CÍA. S.R.L.   
27484       2024-09       3399  33-69534030-9    PEGORIN Y CÍA. S.R.L.   

                                          direccion localidad   provincia  \
33945  REMEDIOS DE ESCALADA 1782 DORREGO GUAYMALLEN   DORREGO     MENDOZA   
33944  REMEDIOS DE ESCALADA 1782 DORREGO GUAYMALLEN   DORREGO     MENDOZA   
33942  REMEDIOS DE ESCALADA 1782 DORREGO GUAYMALLEN   DORREGO     MENDOZA   
33943  REMEDIOS DE ESCALADA 1782 DORREGO GUAYMALLEN   DORREGO     MENDOZA   
11254                      RUTA  NACIONAL 12 KM 500  EL PINGO  ENTRE RIOS   
27485                                RUTA 14 KM. 13    PIÑERO    SANTA FE   
27486                                RUTA 14 KM. 13    PIÑERO    SANTA FE   
27488                                RUTA 14 KM. 13    PIÑERO    SANTA FE   
27487                                RUTA 14 KM. 13    PIÑERO    SANTA FE   
27484                                RUTA 14 KM. 13    PIÑERO    SANTA FE   

       region  idproducto                         producto  idtipohorario  \
33945    CUYO          21                  Gas Oil Grado 3              3   
33944    CUYO          21                  Gas Oil Grado 3              2   
33942    CUYO          19                  Gas Oil Grado 2              2   
33943    CUYO          19                  Gas Oil Grado 2              3   
11254     NaN          19                  Gas Oil Grado 2              3   
27485  CENTRO          21                  Gas Oil Grado 3              2   
27486  CENTRO          21                  Gas Oil Grado 3              3   
27488  CENTRO           2  Nafta (súper) entre 92 y 95 Ron              3   
27487  CENTRO           2  Nafta (súper) entre 92 y 95 Ron              2   
27484  CENTRO          19                  Gas Oil Grado 2              3   

      tipohorario  precio       fecha_vigencia  idempresabandera  \
33945    Nocturno  2000.0  2024-08-01 11:35:00                 1   
33944      Diurno  2000.0  2024-08-01 11:35:00                 1   
33942      Diurno  1999.0  2024-07-02 17:06:00                 1   
33943    Nocturno  1999.0  2024-07-02 17:06:00                 1   
11254    Nocturno  1998.0  2024-09-01 00:00:00                 2   
27485      Diurno  1850.0  2024-09-10 15:00:00                 1   
27486    Nocturno  1850.0  2024-09-10 15:00:00                 1   
27488    Nocturno  1800.0  2024-09-10 15:00:00                 1   
27487      Diurno  1800.0  2024-09-10 15:00:00                 1   
27484    Nocturno  1700.0  2024-09-10 15:00:00                 1   

      empresabandera    latitud   longitud  \
33945         BLANCA -32.912625 -68.833215   
33944         BLANCA -32.912625 -68.833215   
33942         BLANCA -32.912625 -68.833215   
33943         BLANCA -32.912625 -68.833215   
11254            YPF -31.567460 -59.960210   
27485         BLANCA -33.095440 -60.793440   
27486         BLANCA -33.095440 -60.793440   
27488         BLANCA -33.095440 -60.793440   
27487         BLANCA -33.095440 -60.793440   
27484         BLANCA -33.095440 -60.793440   

                                                 geojson  
33945  {"type":"Point","coordinates":[-68.833215,-32....  
33944  {"type":"Point","coordinates":[-68.833215,-32....  
33942  {"type":"Point","coordinates":[-68.833215,-32....  
339

Los precios más altos se encuentran las provincias de **Mendoza, Entre Ríos y Santa Fe.**

In [56]:
round(df_precios_combustible.groupby("provincia")["precio"].mean(), 2).sort_values(ascending = True)

provincia
LA RIOJA                910.34
TIERRA DEL FUEGO        983.12
MENDOZA                1000.30
JUJUY                  1033.03
TUCUMAN                1041.41
CHUBUT                 1052.83
SALTA                  1056.49
SANTA CRUZ             1058.60
LA PAMPA               1064.08
RIO NEGRO              1067.16
SANTA FE               1073.81
SANTIAGO DEL ESTERO    1077.18
FORMOSA                1081.64
NEUQUEN                1088.43
CAPITAL FEDERAL        1097.68
CORDOBA                1104.60
SAN JUAN               1111.53
BUENOS AIRES           1114.45
SAN LUIS               1138.65
CATAMARCA              1173.89
ENTRE RIOS             1177.91
CHACO                  1223.27
MISIONES               1268.16
CORRIENTES             1270.21
Name: precio, dtype: float64

En promedio, los precios más bajos se encuentran las provincias de **La Rioja, Tierra del Fuego y Mendoza.**

In [24]:
df_precios_combustible.sort_values("precio", ascending = True).head(10)

indice_tiempo  idempresa           cuit  \
1437        2017-04       3220  30-64192181-1   
23582       2017-03       2821  30-70164211-9   
23581       2017-03       2821  30-70164211-9   
17727       2017-01       5657  30-61045648-7   
17728       2017-01       5657  30-61045648-7   
33652       2017-03       3727  30-70844148-8   
33653       2017-03       3727  30-70844148-8   
7039        2017-01       4651  30-69137550-8   
7040        2017-01       4651  30-69137550-8   
10265       2017-03       2919  30-70823453-9   

                                           empresa               direccion  \
1437                     ALVAREZ JONTE 6296 S.R.L.      ALVAREZ JONTE 6296   
23582              MONTES G-GARGIULO ANTONIO  S.H.                60 Y 150   
23581              MONTES G-GARGIULO ANTONIO  S.H.                60 Y 150   
17727                  H Y G L CONSTRUCCIONES S.A.  Avenida Bme Mitre 4708   
17728                  H Y G L CONSTRUCCIONES S.A.  Avenida Bme Mitre 4708   
33652                            SERVISOL 131 S.A.                131 1820   
33653                            SERVISOL 131 S.A.                131 1820   
7039   COMBUSTIBLES BOULEVARD SETENTA Y DOS S.R.L.                 72 1194   
7040   COMBUSTIBLES BOULEVARD SETENTA Y DOS S.R.L.                 72 1194   
10265                                     DOKAN SA    RUTA NAC N° 7 KM 140   

             localidad        provincia    region  idproducto producto  \
1437   CAPITAL FEDERAL  CAPITAL FEDERAL  PAMPEANA           6      GNC   
23582         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
23581         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
17727          CASEROS     BUENOS AIRES  PAMPEANA           6      GNC   
17728          CASEROS     BUENOS AIRES  PAMPEANA           6      GNC   
33652         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
33653         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
7039          LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
7040          LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
10265  CARMEN DE ARECO     BUENOS AIRES  PAMPEANA           6      GNC   

       idtipohorario tipohorario  precio       fecha_vigencia  \
1437               3    Nocturno   6.999  2017-04-07 06:00:00   
23582              3    Nocturno   7.000  2017-03-23 12:43:00   
23581              2      Diurno   7.000  2017-03-23 12:43:00   
17727              2      Diurno   7.777  2017-01-11 14:00:00   
17728              3    Nocturno   7.777  2017-01-11 14:00:00   
33652              2      Diurno   8.000  2017-03-23 12:46:00   
33653              3    Nocturno   8.000  2017-03-23 12:46:00   
7039               2      Diurno   8.290  2017-01-20 16:39:00   
7040               3    Nocturno   8.290  2017-01-20 16:39:00   
10265              2      Diurno   8.899  2017-03-30 16:50:00   

       idempresabandera empresabandera    latitud   longitud  \
1437                  1         BLANCA -34.632822 -58.520051   
23582                28           PUMA -34.961590 -57.986040   
23581                28           PUMA -34.961590 -57.986040   
17727                 2            YPF -34.610180 -58.562540   
17728                 2            YPF -34.610180 -58.562540   
33652                26          AXION -34.953085 -57.952810   
33653                26          AXION -34.953085 -57.952810   
7039                  2            YPF -34.942640 -57.939440   
7040                  2            YPF -34.942640 -57.939440   
10265                 1         BLANCA -34.403207 -59.836778   

                                                 geojson  
1437   {"type":"Point","coordinates":[-58.520051,-34....  
23582  {"type":"Point","coordinates":[-57.98604,-34.9...  
23581  {"type":"Point","coordinates":[-57.98604,-34.9...  
17727  {"type":"Point","coordinates":[-58.56254,-34.6...  
17728  {"type":"Point","coordinates":[-58.56254,-34.6...  
33652  {"type":"P

En cuanto a los combustibles más asequibles, se pueden conseguir en la **Zona de Capital Federal y Provincia de Buenos Aires.**

### Inciso b

In [25]:
round(df_precios_combustible.groupby("producto")["precio"].mean(), 2)

producto
GNC                                  487.02
Gas Oil Grado 2                     1085.86
Gas Oil Grado 3                     1305.45
Nafta (premium) de más de 95 Ron    1267.47
Nafta (súper) entre 92 y 95 Ron     1044.29
Name: precio, dtype: float64

### Inciso c

In [26]:
df_precios_combustible.sort_values("precio", ascending=True).head(10)

indice_tiempo  idempresa           cuit  \
1437        2017-04       3220  30-64192181-1   
23582       2017-03       2821  30-70164211-9   
23581       2017-03       2821  30-70164211-9   
17727       2017-01       5657  30-61045648-7   
17728       2017-01       5657  30-61045648-7   
33652       2017-03       3727  30-70844148-8   
33653       2017-03       3727  30-70844148-8   
7039        2017-01       4651  30-69137550-8   
7040        2017-01       4651  30-69137550-8   
10265       2017-03       2919  30-70823453-9   

                                           empresa               direccion  \
1437                     ALVAREZ JONTE 6296 S.R.L.      ALVAREZ JONTE 6296   
23582              MONTES G-GARGIULO ANTONIO  S.H.                60 Y 150   
23581              MONTES G-GARGIULO ANTONIO  S.H.                60 Y 150   
17727                  H Y G L CONSTRUCCIONES S.A.  Avenida Bme Mitre 4708   
17728                  H Y G L CONSTRUCCIONES S.A.  Avenida Bme Mitre 4708   
33652                            SERVISOL 131 S.A.                131 1820   
33653                            SERVISOL 131 S.A.                131 1820   
7039   COMBUSTIBLES BOULEVARD SETENTA Y DOS S.R.L.                 72 1194   
7040   COMBUSTIBLES BOULEVARD SETENTA Y DOS S.R.L.                 72 1194   
10265                                     DOKAN SA    RUTA NAC N° 7 KM 140   

             localidad        provincia    region  idproducto producto  \
1437   CAPITAL FEDERAL  CAPITAL FEDERAL  PAMPEANA           6      GNC   
23582         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
23581         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
17727          CASEROS     BUENOS AIRES  PAMPEANA           6      GNC   
17728          CASEROS     BUENOS AIRES  PAMPEANA           6      GNC   
33652         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
33653         LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
7039          LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
7040          LA PLATA     BUENOS AIRES  PAMPEANA           6      GNC   
10265  CARMEN DE ARECO     BUENOS AIRES  PAMPEANA           6      GNC   

       idtipohorario tipohorario  precio       fecha_vigencia  \
1437               3    Nocturno   6.999  2017-04-07 06:00:00   
23582              3    Nocturno   7.000  2017-03-23 12:43:00   
23581              2      Diurno   7.000  2017-03-23 12:43:00   
17727              2      Diurno   7.777  2017-01-11 14:00:00   
17728              3    Nocturno   7.777  2017-01-11 14:00:00   
33652              2      Diurno   8.000  2017-03-23 12:46:00   
33653              3    Nocturno   8.000  2017-03-23 12:46:00   
7039               2      Diurno   8.290  2017-01-20 16:39:00   
7040               3    Nocturno   8.290  2017-01-20 16:39:00   
10265              2      Diurno   8.899  2017-03-30 16:50:00   

       idempresabandera empresabandera    latitud   longitud  \
1437                  1         BLANCA -34.632822 -58.520051   
23582                28           PUMA -34.961590 -57.986040   
23581                28           PUMA -34.961590 -57.986040   
17727                 2            YPF -34.610180 -58.562540   
17728                 2            YPF -34.610180 -58.562540   
33652                26          AXION -34.953085 -57.952810   
33653                26          AXION -34.953085 -57.952810   
7039                  2            YPF -34.942640 -57.939440   
7040                  2            YPF -34.942640 -57.939440   
10265                 1         BLANCA -34.403207 -59.836778   

                                                 geojson  
1437   {"type":"Point","coordinates":[-58.520051,-34....  
23582  {"type":"Point","coordinates":[-57.98604,-34.9...  
23581  {"type":"Point","coordinates":[-57.98604,-34.9...  
17727  {"type":"Point","coordinates":[-58.56254,-34.6...  
17728  {"type":"Point","coordinates":[-58.56254,-34.6...  
33652  {"type":"P

La empresa **ALVAREZ JONTE 6296 S.R.L.**, ubicada en Capital Federal, es la que más barato comercializa el GNC, a tan sólo **$6.99** el M3

### inciso d

In [27]:
df_nafta = df_precios_combustible[df_precios_combustible['producto'].str.contains('Nafta')]
df_nafta

indice_tiempo  idempresa           cuit               empresa  \
6           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
7           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
8           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
9           2024-09       1376  33-64337382-9  10 DE SETIEMBRE S.A.   
14          2024-08      10677  30-71752323-3           1263 S.R.L.   
...             ...        ...            ...                   ...   
36713       2024-09       6459  33-70824089-9         ZONA CERO S.A   
36718       2024-09       4097  30-70795820-7           ZUSICO S.A.   
36719       2024-09       4097  30-70795820-7           ZUSICO S.A.   
36720       2024-09       4097  30-70795820-7           ZUSICO S.A.   
36721       2024-09       4097  30-70795820-7           ZUSICO S.A.   

                                               direccion           localidad  \
6                                        Av. Mosconi 299   LOMAS DEL MIRADOR   
7                                        Av. Mosconi 299   LOMAS DEL MIRADOR   
8                                        Av. Mosconi 299   LOMAS DEL MIRADOR   
9                                        Av. Mosconi 299   LOMAS DEL MIRADOR   
14     UTA NAC. 3 KM. 1258 (COLECTORA CALLE 102 Nº 32...       SIERRA GRANDE   
...                                                  ...                 ...   
36713                                    RIVADAVIA Nº 38  COMODORO RIVADAVIA   
36718                                    TALCAHUANO 1198       MAR DEL PLATA   
36719                                    TALCAHUANO 1198       MAR DEL PLATA   
36720                                    TALCAHUANO 1198       MAR DEL PLATA   
36721                                    TALCAHUANO 1198       MAR DEL PLATA   

          provincia     region  idproducto                          producto  \
6      BUENOS AIRES   PAMPEANA           3  Nafta (premium) de más de 95 Ron   
7      BUENOS AIRES   PAMPEANA           3  Nafta (premium) de más de 95 Ron   
8      BUENOS AIRES   PAMPEANA           2   Nafta (súper) entre 92 y 95 Ron   
9      BUENOS AIRES   PAMPEANA           2   Nafta (súper) entre 92 y 95 Ron   
14        RIO NEGRO        NaN           3  Nafta (premium) de más de 95 Ron   
...             ...        ...         ...                               ...   
36713        CHUBUT  PATAGONIA           2   Nafta (súper) entre 92 y 95 Ron   
36718  BUENOS AIRES   PAMPEANA           3  Nafta (premium) de más de 95 Ron   
36719  BUENOS AIRES   PAMPEANA           3  Nafta (premium) de más de 95 Ron   
36720  BUENOS AIRES   PAMPEANA           2   Nafta (súper) entre 92 y 95 Ron   
36721  BUENOS AIRES   PAMPEANA           2   Nafta (súper) entre 92 y 95 Ron   

       idtipohorario tipohorario  precio       fecha_vigencia  \
6                  2      Diurno  1371.0  2024-09-07 10:52:00   
7                  3    Nocturno  1371.0  2024-09-07 10:52:00   
8                  2      Diurno  1119.0  2024-09-07 10:53:00   
9                  3    Nocturno  1119.0  2024-09-07 10:53:00   
14                 2      Diurno  1041.0  2024-08-01 16:56:00   
...              ...         ...     ...                  ...   
36713              3    Nocturno   872.0  2024-09-02 15:28:00   
36718              2      Diurno  1416.0  2024-09-01 09:00:00   
36719              3    Nocturno  1416.0  2024-09-01 09:00:00   
36720              2      Diurno  1178.0  2024-09-01 09:00:00   
36721              3    Nocturno  1178.0  2024-09-01 09:00:00   

       idempresabandera empresabandera    latitud   longitud  \
6                    28           PUMA -34.658476 -58.529443   
7                    28           PUMA -34.658476 -58.529443   
8                    28           PUMA -34.658476 -58.529443   
9                    28           PUMA -34.658476 -58.529443   
14                   28           PUMA -41.607070 -65.353830   
...                 ...            ...        ...        ...   
36

In [54]:
df_nafta_por_provincia = round(df_nafta.groupby("provincia")["precio"].mean(), 2)
df_nafta_por_provincia

empresa
10 DE SETIEMBRE S.A.    1245.0
1263 S.R.L.              935.0
1913 SRL                 934.0
197 COMBUSTIBLES SRL    1182.0
25 ALVEAR S.R.L         1280.0
                         ...  
ZOKFA S.A.S.            1268.0
ZONA CERO S.A           1000.0
ZUSICO S.A.             1297.0
ÑANDU S.R.L.            1305.0
ÑUSCA S.A.               969.5
Name: precio, Length: 3159, dtype: float64

## Ejercicio 6

In [29]:
df_ejerc_6 = pd.read_csv(path_folder + "/share_mujer_todos.csv")
df_puestos_mujeres = df_ejerc_6.copy()
df_puestos_mujeres

fecha            zona_prov  share_mujer
0     2013-01-01         BUENOS AIRES     0.396839
1     2013-01-01      CAPITAL FEDERAL     0.405973
2     2013-01-01            CATAMARCA     0.387500
3     2013-01-01                CHACO     0.396491
4     2013-01-01               CHUBUT     0.363042
...          ...                  ...          ...
3139  2023-11-01           SANTA CRUZ     0.324173
3140  2023-11-01             SANTA FE     0.407061
3141  2023-11-01  SANTIAGO DEL ESTERO     0.402643
3142  2023-11-01     TIERRA DEL FUEGO     0.434650
3143  2023-11-01              TUCUMAN     0.375821

[3144 rows x 3 columns]

In [30]:
df_puestos_mujeres.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3144 entries, 0 to 3143
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   fecha        3144 non-null   object 
 1   zona_prov    3144 non-null   object 
 2   share_mujer  3144 non-null   float64
dtypes: float64(1), object(2)
memory usage: 73.8+ KB


### Inciso a

Forma que se me ocurrió a **mí**

In [31]:
df_fecha_separada = df_puestos_mujeres.copy()
df_fecha_separada[["año", "mes", "dia"]] = df_puestos_mujeres["fecha"].str.split("-", expand=True)
df_fecha_separada

fecha            zona_prov  share_mujer   año mes dia
0     2013-01-01         BUENOS AIRES     0.396839  2013  01  01
1     2013-01-01      CAPITAL FEDERAL     0.405973  2013  01  01
2     2013-01-01            CATAMARCA     0.387500  2013  01  01
3     2013-01-01                CHACO     0.396491  2013  01  01
4     2013-01-01               CHUBUT     0.363042  2013  01  01
...          ...                  ...          ...   ...  ..  ..
3139  2023-11-01           SANTA CRUZ     0.324173  2023  11  01
3140  2023-11-01             SANTA FE     0.407061  2023  11  01
3141  2023-11-01  SANTIAGO DEL ESTERO     0.402643  2023  11  01
3142  2023-11-01     TIERRA DEL FUEGO     0.434650  2023  11  01
3143  2023-11-01              TUCUMAN     0.375821  2023  11  01

[3144 rows x 6 columns]

In [32]:
top_años_ocup = round(df_fecha_separada.groupby("año")["share_mujer"].mean(), 4)
top_años_ocup.sort_values(ascending=False)

año
2023    0.4069
2020    0.3993
2022    0.3991
2021    0.3990
2019    0.3954
2018    0.3920
2017    0.3895
2016    0.3879
2015    0.3813
2014    0.3801
2013    0.3755
Name: share_mujer, dtype: float64

Forma eficiente con funciones de pandas **(DatetimeIndex)**

In [33]:
df_ord_fechas = df_puestos_mujeres.copy()
df_ord_fechas['year'] = pd.DatetimeIndex(df_ord_fechas['fecha']).year
df_ord_fechas

fecha            zona_prov  share_mujer  year
0     2013-01-01         BUENOS AIRES     0.396839  2013
1     2013-01-01      CAPITAL FEDERAL     0.405973  2013
2     2013-01-01            CATAMARCA     0.387500  2013
3     2013-01-01                CHACO     0.396491  2013
4     2013-01-01               CHUBUT     0.363042  2013
...          ...                  ...          ...   ...
3139  2023-11-01           SANTA CRUZ     0.324173  2023
3140  2023-11-01             SANTA FE     0.407061  2023
3141  2023-11-01  SANTIAGO DEL ESTERO     0.402643  2023
3142  2023-11-01     TIERRA DEL FUEGO     0.434650  2023
3143  2023-11-01              TUCUMAN     0.375821  2023

[3144 rows x 4 columns]

In [34]:
top_años_ocup = round(df_ord_fechas.groupby("year")["share_mujer"].mean(), 4)
top_años_ocup.sort_values(ascending=False)

year
2023    0.4069
2020    0.3993
2022    0.3991
2021    0.3990
2019    0.3954
2018    0.3920
2017    0.3895
2016    0.3879
2015    0.3813
2014    0.3801
2013    0.3755
Name: share_mujer, dtype: float64

El año **2023** fue el que tuvo más ocupación femenina, con un promedio de **40.69%**.

In [35]:
df_puestos_mujeres.sort_values("share_mujer", ascending=False).head()

fecha zona_prov  share_mujer
3123  2023-11-01     CHACO     0.451039
3099  2023-10-01     CHACO     0.447035
2067  2020-03-01     CHACO     0.445126
1995  2019-12-01     CHACO     0.445041
2506  2021-09-01  LA PAMPA     0.444698

Siendo más específicos, en **Noviembre de 2023** tuvo lugar el mayor porcentaje de mujeres registradas ocupadas con un **45.10%** en la provincia de Chaco.

### Inciso b

In [36]:
mujeres_ocupadas_porc_hist = round(df_puestos_mujeres.groupby("zona_prov")["share_mujer"].mean(), 4)
mujeres_ocupadas_porc_hist.sort_values(ascending=False)

zona_prov
LA PAMPA               0.4281
JUJUY                  0.4200
CHACO                  0.4170
CAPITAL FEDERAL        0.4130
LA RIOJA               0.4112
RIO NEGRO              0.4098
TIERRA DEL FUEGO       0.4093
ENTRE RIOS             0.4089
BUENOS AIRES           0.4069
CORDOBA                0.4030
SANTIAGO DEL ESTERO    0.3990
FORMOSA                0.3973
SAN LUIS               0.3958
CATAMARCA              0.3954
MENDOZA                0.3925
SALTA                  0.3907
SANTA FE               0.3889
MISIONES               0.3856
NEUQUEN                0.3827
CHUBUT                 0.3818
SAN JUAN               0.3678
CORRIENTES             0.3667
TUCUMAN                0.3583
SANTA CRUZ             0.2627
Name: share_mujer, dtype: float64

La provincia con mayor porcentaje de mujeres ocupadas a lo largo de los años es **La Pampa**, con un promedio de **42, 81%** de ocupación femenina en la provincia, seguida por **Jujuy** con un **42,00%**.

### Inciso c

In [37]:
top_meses_ocup = round(df_fecha_separada.groupby("mes")["share_mujer"].mean(), 4)
top_meses_ocup.sort_values(ascending=False)

mes
12    0.3936
11    0.3934
10    0.3933
09    0.3923
08    0.3921
06    0.3919
07    0.3914
01    0.3907
05    0.3905
04    0.3896
02    0.3895
03    0.3880
Name: share_mujer, dtype: float64

El mes que en promedio mayor ocupación posee es **Diciembre**

### Inciso d

In [38]:
df_ocup_mayor_50 = df_puestos_mujeres[df_puestos_mujeres["share_mujer"] > 0.5]
df_ocup_mayor_50

Empty DataFrame
Columns: [fecha, zona_prov, share_mujer]
Index: []

En **ningún caso** la ocupación femenina superó el 50% del total de empleos registrados.

In [39]:
df_ocup_mayor_40 = df_puestos_mujeres[df_puestos_mujeres["share_mujer"] > 0.4]
df_ocup_mayor_40

fecha            zona_prov  share_mujer
1     2013-01-01      CAPITAL FEDERAL     0.405973
5     2013-01-01              CORDOBA     0.403906
9     2013-01-01                JUJUY     0.403949
10    2013-01-01             LA PAMPA     0.424298
11    2013-01-01             LA RIOJA     0.404830
...          ...                  ...          ...
3137  2023-11-01             SAN JUAN     0.400000
3138  2023-11-01             SAN LUIS     0.429164
3140  2023-11-01             SANTA FE     0.407061
3141  2023-11-01  SANTIAGO DEL ESTERO     0.402643
3142  2023-11-01     TIERRA DEL FUEGO     0.434650

[1390 rows x 3 columns]

In [40]:
print(f"La ocupación femenina superó el 40% en el {round(len(df_ocup_mayor_40) / len(df_puestos_mujeres), 4) * 100}% de los casos.")

La ocupación femenina superó el 40% en el 44.21% de los casos.
